sentry_sdk/integrations/anthropic.py

```diff
@@ -0,0 +1,179 @@
+from functools import wraps
+
+import sentry_sdk
+from sentry_sdk.consts import OP
+from sentry_sdk.integrations import DidNotEnable, Integration
+from sentry_sdk.scope import should_send_default_pii
+from sentry_sdk.utils import (
+    capture_internal_exceptions,
+    ensure_integration_enabled,
+    event_from_exception,
+    package_version,
+)
+
+from anthropic.resources import Messages
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any, Iterator
+    from anthropic.types import MessageStreamEvent
+    from sentry_sdk.tracing import Span
+
+
+COMPLETION_TOKENS_USED = "ai.completion_tоkens.used"
+PROMPT_TOKENS_USED = "ai.prompt_tоkens.used"
+TOTAL_TOKENS_USED = "ai.total_tоkens.used"
+
+
+class AnthropicIntegration(Integration):
+    identifier = "anthropic"
+
+    def __init__(self, include_prompts=True):
+        # type: (AnthropicIntegration, bool) -> None
+        self.include_prompts = include_prompts
+
+    @staticmethod
+    def setup_once():
+        # type: () -> None
+        version = package_version("anthropic")
+
+        if version is None:
+            raise DidNotEnable("Unparsable anthropic version.")
+
+        if version < (0, 16):
+            raise DidNotEnable("anthropic 0.16 or newer required.")
+
+        Messages.create = _wrap_message_create(Messages.create)
+
+
+def _capture_exception(exc):
+    # type: (Any) -> None
+    event, hint = event_from_exception(
+        exc,
+        client_options=sentry_sdk.get_client().options,
+        mechanism={"type": "anthropic", "handled": False},
+    )
+    sentry_sdk.capture_event(event, hint=hint)
+
+
+def _calculate_token_usage(result, span):
+    # type: (Messages, Span) -> None
+    input_tokens = 0
+    output_tokens = 0
+    if hasattr(result, "usage"):
+        usage = result.usage
+        if hasattr(usage, "input_tokens") and isinstance(usage.input_tokens, int):
+            input_tokens = usage.input_tokens
+        if hasattr(usage, "output_tokens") and isinstance(usage.output_tokens, int):
+            output_tokens = usage.output_tokens
+
+    total_tokens = input_tokens + output_tokens
+
+    if total_tokens != 0:
+        span.set_data(TOTAL_TOKENS_USED, total_tokens)
+    if input_tokens != 0:
+        span.set_data(PROMPT_TOKENS_USED, input_tokens)
+    if output_tokens != 0:
+        span.set_data(COMPLETION_TOKENS_USED, output_tokens)
```

Comment:
Once langchain is merged, this should use the new helper function https://github.com/getsentry/sentry-python/pull/2911/files#diff-c566959db6701d8426b02bcf30caa0222d01aa9931ce9913b1a35fedfd0953c1R59